### PR TITLE
SAML Authentication content for OpenSearch

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
# What changed, and why it matters
Added content for SAML SSO authentication for Aiven for OpenSearch service. Part of OpenSearch Security beta release.

### CODE REVIEW

File: docs/products/opensearch/howto/saml-sso-authentication.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+To enable SAML authentication for your Aiven for OpenSearch service, follow these steps: 
+
+1. On your Aiven for OpenSearch service, navigate to the **Users** tab.
+2. In the **SAML SSO Authentication** section, select **Enable SAML**. 
+3. On the **Configure SAML Authentication** screen, enter the following details: 
+   
+   * **SSO URL**: This is a distinct URL assigned to each Aiven for OpenSearch service and serves as the destination where the Identity Provider (IdP) sends SAML responses after a user has been authenticated successfully. It is a fixed URL that cannot be modified.
+   * **IdP Metadata URL**: Enter the URL of your SAML Identity Provider's (IdP) metadata that Aiven will use to authenticate users.
+   * **IdP Entity ID**: Enter the unique identifier assigned to your Identity Provider (IdP). This identifier assists Aiven in distinguishing between various IdPs.
+   * **SP Entity ID**: E Enter the unique identifier the IdP uses to recognize and authenticate Aiven for OpenSearch as the service provider.
```
Comment (safa-topal) [inline docs/products/opensearch/howto/saml-sso-authentication.rst:39]: Good to mention here the content of can only be defined by the users themselves. Can be any arbitrary value, OpenSearch docs recommends it's good practice to put the kibana URL here. 

https://opensearch.org/docs/latest/security/authentication-backends/saml/#idp-and-service-provider-entity-id
> We recommend adding a new application for OpenSearch Dashboards and using the URL of your OpenSearch Dashboards installation as the SP entity ID.

File: docs/products/opensearch/howto/saml-sso-authentication.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+---------------
+* Aiven for OpenSearch® version 2.4 or later is required. If you are using an earlier version, upgrade to the latest version.
+* OpenSearch Security management must be enabled on the Aiven for OpenSearch® service.
+* You will need a SAML identity provider (IdP), the Metadata URL, and IdP entity ID.
+
+
+Configure SAML on IdP
+---------------------
+
+Enabling SAML SSO Authentication for Aiven for OpenSearch requires the configuration of SAML with an Identity Provider (IdP). Since Aiven for OpenSearch supports multiple IdPs, the steps for configuring SAML with your IdP may differ depending on the IdP you are using. It is recommended to refer to your identity provider's documentation for detailed instructions on configuring SAML applications.
```
[the remark this answers is not in the document]
Comment (safa-topal) [reply docs/products/opensearch/howto/saml-sso-authentication.rst:19]: Probably my wordings aren't as good as yours, so whichever way you want to amend it please go forward.

**Review by safa-topal — CHANGES_REQUESTED** [review]
LGTM overall, a few change requests to improve accuracy of statements and a nitpick.

File: docs/products/opensearch/howto/saml-sso-authentication.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (harshini-rangaswamy) [reply docs/products/opensearch/howto/saml-sso-authentication.rst]: Modified the text: As Aiven for OpenSearch is designed to work with various SAML IdPs, the configuration steps may differ depending on your IdP. Refer to your IdP's documentation for detailed instructions on configuring SAML applications.

File: docs/products/opensearch/howto/saml-sso-authentication.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (harshini-rangaswamy) [reply docs/products/opensearch/howto/saml-sso-authentication.rst]: fixed this

**Review by safa-topal — APPROVED** [review]

**Pull request merged**